packages/kbn-config-schema/src/types/type.ts

```diff
@@ -214,16 +222,62 @@ export abstract class Type<V> {
   }
 }
 
-function recursiveGetSchemaStructure(internalSchema: AnySchema, path: string[] = []) {
+function recursiveGetSchemaStructure(internalSchema: Schema, path: string[] = []) {
   const array: SchemaStructureEntry[] = [];
   // Note: we are relying on Joi internals to obtain the schema structure (recursive keys).
   // This is not ideal, but it works for now and we only need it for some integration test assertions.
   // If it breaks in the future, we'll need to update our tests.
   for (const [key, val] of (internalSchema as any)._ids._byKey.entries()) {
     array.push(...recursiveGetSchemaStructure(val.schema, [...path, key]));
   }
+
   if (!array.length) {
-    array.push({ path, type: internalSchema.type ?? 'unknown' });
+    let type: string;
+    try {
+      type = [...new Set([getSpecialType(internalSchema, false, path)].flat())]
+        .filter(Boolean)
+        .join('|');
+    } catch (error) {
+      // failed to find special type, might need to update for new joi versions or type usages
+      type = internalSchema.type || 'unknown';
+    }
+
+    array.push({
+      path,
+      type,
+    });
   }
   return array;
 }
+
+function getSpecialType(
```

Comment:
nit: Can we add a small comment describing what this fn does? Just input => output.

Also WDYT of naming this `prettyPrintType`? Naming amirite

Reply:
Yeah that's fine with me, see f599961df97dfa39cbc8e8ab0ba90bb253221dd1.